test_coco.py

```diff
@@ -266,15 +270,59 @@ def test(data,
             cocoEval.accumulate()
             cocoEval.summarize()
             map, map50 = cocoEval.stats[:2]  # update results (mAP@0.5:0.95, mAP@0.5)
+            
+            if evaluate_fbeta:
+                cocoEval.accumulateFBeta()
+                cocoEval.summarizeFBetaScores(average=fbeta_average_method)
+                f1_score, _, _, _ = cocoEval.getBestFBeta(beta=1, iouThr=iou_thres, average=fbeta_average_method)
+                f2_score, _, _, _ = cocoEval.getBestFBeta(beta=2, iouThr=iou_thres, average=fbeta_average_method)
+                print(f"Results:")
+                s = ('%20s ' * 4) % ('f1-score', 'f2-score', 'map', 'map50')
+                print(f"{s}")
+                s = ('%20s ' * 4) % (f1_score, f2_score, map, map50)
+                print(f"{s}")
+                
+                evaluate_fbeta_success = True
+
+            if results_csv:
+                import csv
+                
+                result = [save_dir, f1_score, f2_score, map, map50]
+                if os.path.isfile(results_csv):
+                    # append to csv
+                    with open(results_csv, 'a', encoding='UTF8') as f:
+                        writer = csv.writer(f)
+                        writer.writerow(result)
+                    print(f"Results appended to existing CSV file: {results_csv}")
+                else:
+                    # create new csv
+                    with open(results_csv, 'w', encoding='UTF8') as f:
+                        writer = csv.writer(f)
+                        header = ['save dir', 'f1', 'f2', 'map', 'map50']
+                        writer.writerow(header)
+                        writer.writerow(result)
+                        print(f"New CSV file created: {results_csv} and results added")
+        
         except Exception as e:
-            print(f'pycocotools unable to run: {e}')
+            if isinstance(e, IsADirectoryError):
+                print(
+                    f"ERROR: Invalid data format for pycocotools evaluation.\n"
+                    f"\tThe provided 'data' YAML file indicates a directory, but a .json file is required.\n"
+                    f"\tPlease provide the path to a valid .json file instead of a directory to evaluate with pycocotools.\n"
+                    f"\tError Details: {e}"
+                )
+            else:
+                print(f'ERROR: pycocotools unable to run:\n\t{e}')
+            
 
     # Return results
     s = f"\n{len(list(save_dir.glob('labels/*.txt')))} labels saved to {save_dir / 'labels'}" if save_txt else ''
     print(f"Results saved to {save_dir}{s}")
     maps = np.zeros(nc) + map
     for i, c in enumerate(ap_class):
         maps[c] = ap[i]
+    if evaluate_fbeta_success:
+        return (f1_score, f2_score, mp, mr, map50, map, *(loss.cpu() / len(dataloader)).tolist()), maps, t
     return (mp, mr, map50, map, *(loss.cpu() / len(dataloader)).tolist()), maps, t
```

Comment:
Hmm actually for this code the return function does nothing hor. Like the results are not used anywhere